commit 849ff5353023c95d40040ecfd279e216074b3d1c
Author: popcornmix <popcornmix@gmail.com>
Date:   Wed Jun 18 13:42:01 2014 +0100

    vmstat: Workaround for issue where dirty page count goes negative
    
    See:
    https://github.com/raspberrypi/linux/issues/617
    http://www.spinics.net/lists/linux-mm/msg72236.html

diff --git a/include/linux/vmstat.h b/include/linux/vmstat.h
index 3e5d907..2539068 100644
--- a/include/linux/vmstat.h
+++ b/include/linux/vmstat.h
@@ -219,7 +219,11 @@ static inline void __inc_zone_state(struct zone *zone, enum zone_stat_item item)
 static inline void __dec_zone_state(struct zone *zone, enum zone_stat_item item)
 {
 	atomic_long_dec(&zone->vm_stat[item]);
+	if (item == NR_FILE_DIRTY && unlikely(atomic_long_read(&zone->vm_stat[item]) < 0))
+		atomic_long_set(&zone->vm_stat[item], 0);
 	atomic_long_dec(&vm_stat[item]);
+	if (item == NR_FILE_DIRTY && unlikely(atomic_long_read(&vm_stat[item]) < 0))
+		atomic_long_set(&vm_stat[item], 0);
 }
 
 static inline void __inc_zone_page_state(struct page *page,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1002_849ff5353023c95d40040ecfd279e216074b3d1c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1002_849ff5353023c95d40040ecfd279e216074b3d1c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
